app/forms/questionnaire_form.py

```diff
@@ -386,8 +386,24 @@ def _get_period_limits(
         return minimum, maximum
 
     def _get_formatted_calculation_values(
-        self, answers_list: Sequence[str]
+        self, answers_sequence: Sequence[str]
     ) -> list[str]:
+        answers_list = list(answers_sequence)
+        block_id = self.location.block_id if self.location else None
+        if block_id and self.schema.block_has_dynamic_answer(block_id):
+            list_name = (
+                self.schema.get_list_name_for_dynamic_answer(block_id)
+                if block_id
+                else None
+            )
+            list_item_ids = self.list_store[list_name] if list_name else None
+            for answer_id in answers_list:
```

Comment:
I'd avoid extending a list whilst iterating over it, and suggest changing this so that instead of making answers_list as a copy of answers sequence at the top, instead initialise it as an empty list, and then iterate over the sequence and add values based off that, ie.
```
answers_list = []
... 
for answer_index, answer_id in answers_sequence:
   if self.schema.is_answer_dynamic(answer_id) and list_item_ids:
     answers_list.extend(f"{answer_id}-{list_item_id}" for list_item_id in list_item_ids)
   else:
     answers_list.append(answer_id)

Reply:
Well, this needs to take into account scenarios when block doesn't have dynamic answers lol, otherwise it works great and I agree this is better. We can't remove replace since it is removing number separators from our answers?

Reply:
yeah good point - your solution will definitely be better for the no dynamic answers case - and now you've got best of both, nice 👍 
so yeah the .replace comment doesn't make sense anymore